src/Microsoft.Diagnostics.Runtime/Utilities/DumpReader.cs

```diff
@@ -2522,4 +2721,350 @@ internal enum ProcessorArchitecture : ushort
+        }
+    }
+
+    /// <summary>
+    /// MiniDumpHandle type enumeration
+    /// </summary>
+    public enum DumpHandleType
+    {
+        /// <summary>
+        /// No Type
+        /// </summary>
+        NONE,
+        /// <summary>
+        /// Thread Type
+        /// </summary>
+        THREAD,
+        /// <summary>
+        /// Mutex Type
+        /// </summary>
+        MUTEX1,
+        /// <summary>
+        /// Mutex Type
+        /// </summary>
+        MUTEX2,
+        /// <summary>
+        /// Process Type
+        /// </summary>
+        PROCESS1,
+        /// <summary>
+        /// Process Type
+        /// </summary>
+        PROCESS2,
+        /// <summary>
+        /// Event Type
+        /// </summary>
+        EVENT,
+        /// <summary>
+        /// Critical Section Type
+        /// </summary>
+        SECTION,
+        /// <summary>
+        /// Max Type
+        /// </summary>
+        TYPE_MAX
+    }
+    /// <summary>
+    /// Mutex unknown fields
+    /// </summary>
+    public class MutexUnknownFields
+    {
+        /// <summary>
+        /// First mutex unknown field
+        /// </summary>
+        public object Field1 { get; internal set; }
+        /// <summary>
+        /// Second mutex unknown field
+        /// </summary>
+        public object Field2 { get; internal set; }
+    }
+
+    /// <summary>
+    /// Handle additional object info constructed by MINIDUMP_HANDLE_OBJECT_INFORMATION structure 
+    /// </summary>
+    public class DumpHandleInfo
+    {
+        internal DumpHandleInfo(MINIDUMP_HANDLE_OBJECT_INFORMATION info)
+        {
+            this.SizeOfInfo = info.SizeOfInfo;
+            this.InfoType = (DumpHandleType)info.InfoType;
+        }
+        /// <summary>
+        /// Type of object 
+        /// </summary>
+        internal DumpHandleType InfoType { get; private set; }
+
+        /// <summary>
+        /// Size of object information
+        /// </summary>
+        internal UInt32 SizeOfInfo { get; private set; }
+    }
+
+#if !(_WIN64)
+
+    internal struct THREAD_ADDITIONAL_INFO
+    {
+        public UInt32 Unknown1;
+        public UInt32 Unknown2;
+        public UInt32 ProcessId;
+        public UInt32 ThreadId;
+        public UInt32 Unknown3;
+        public UInt32 Priority;
+        public UInt32 Unknown4;
+    };
+
+    internal struct MUTEX_ADDITIONAL_INFO_1
+    {
+        public UInt32 Unknown1;
+        public UInt32 Unknown2;
+    };
+
+    internal struct MUTEX_ADDITIONAL_INFO_2
+    {
+        public UInt32 OwnerProcessId;
+
+        public UInt32 OwnerThreadId;
+    };
+
+    internal struct PROCESS_ADDITIONAL_INFO_2
+    {
+        public UInt32 Unknown1;
+        public UInt32 Unknown2;
+        public UInt32 Unknown3;
+        public UInt32 Unknown4;
+        public UInt32 Unknown5;
+        public UInt32 ProcessId;
+        public UInt32 ParentProcessId;
+        public UInt32 Unknown6;
+    };
+
+#else
+
+    struct MUTEX_ADDITIONAL_INFO_2
```

Comment:
The previous structs are marked explicitly as internal, and these ones aren't. Let's be consistent.